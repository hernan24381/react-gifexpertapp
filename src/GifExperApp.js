import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExperApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

   /*  const handleAdd = () => {
        setCategories([...categories, 'Pepe caret√≥n']);
    } */

    return (
        <>
            <h2>GiftExperApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />



            <ol>
                {
                    categories.map(category => (
                        <GifGrid 
                            key={category}
                            category={category} 
                        />
                    ))
                }
            </ol>


        </>
    )
}

