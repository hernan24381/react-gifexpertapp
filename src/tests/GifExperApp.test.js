import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExperApp } from '../GifExperApp';


describe('pruebas sobre Gif', () => {   
    
    test('debe hacer match con el snapshot', () => {
        
        const wrapper = shallow(<GifExperApp />);
        expect(wrapper).toMatchSnapshot();

    })
    
    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['one punch', 'dragon ball'];
        const wrapper = shallow(<GifExperApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    


})




