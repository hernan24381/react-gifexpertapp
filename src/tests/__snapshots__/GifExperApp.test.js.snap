// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre Gif debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`pruebas sobre Gif debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    GiftExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
